lib/streamlit/elements/media.py

```diff
@@ -133,6 +137,22 @@ def video(
             See https://tools.ietf.org/html/rfc4281 for more info.
         start_time: int
             The time from which this element should start playing.
+        subtitles: str, dict, or io.BytesIO
+            Optional subtitle data for the video, supporting several input types:
+            * None (default): No subtitles.
+            * A string: File path to a subtitle file in '.vtt' or '.srt' formats, or
```

Comment:
That's a good idea!

@kajarenc from a workload POV, is supporting the Python `Path` object doable for the 1.32.0 release? If it's simple enough and fast to implement, let's do it. Else, we can do this as a fast-follow in the subsequent release. WDYT @LukasMasuch?

Reply:
Yes, I think it is a good idea!

Right now we are not super descriptive about the wrong format when there is a typo in the path string (because we do not have a way to easily distinguish between not existing file and just a string with subtitles), but adding a Path type will allow us to raise more specific exception in this case. 

As Lukas mentioned we do not do it in other places in our commands, but we need to start from somewhere, why not start from subtitles :)

Reply:
Sounds good to me! As long we also support passing path strings, we should be good. I have a hunch there'll be less adoption of this feature (especially by novice Python users) if we only support `Path` objects.